spec/hedgelog_spec.rb

```diff
@@ -1,11 +1,300 @@
 require 'spec_helper'
+require 'timecop'
+require 'benchmark'
+require 'json'
 
 describe Hedgelog do
   it 'has a version number' do
     expect(Hedgelog::VERSION).not_to be nil
   end
 
-  it 'does something useful' do
-    expect(false).to eq(true)
+  let(:log_dev) { StringIO.new }
+  let(:log_level) { Logger::INFO }
+  let(:logger) { Hedgelog.new(log_dev) }
+  let(:log_exec) { -> { logger.debug 'foo' } }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.